flang/lib/Lower/Bridge.cpp

```diff
@@ -1281,6 +1281,15 @@ class FirConverter : public Fortran::lower::AbstractConverter {
            "select case list mismatch");
   }
 
+  fir::ExtendedValue
+  genAssociateSelector(const Fortran::semantics::SomeExpr &selector,
+                       Fortran::lower::StatementContext &stmtCtx) {
+    return selector.Rank() > 0 && isArraySectionWithoutVectorSubscript(selector)
```

Comment:
The `selector.Rank() > 0` condition looks odd here.  Name-wise, it looks redundant.  It might best be moved inside `isArraySectionWithoutVectorSubscript`.

Reply:
I was surprised it was needed. There are other places `isArraySectionWithoutVectorSubscript` is used, so I didn't put the test inside out of caution. Thanks for reminding me. I will give it a try and see.